pkgs/development/python-modules/txtai/default.nix

```diff
@@ -0,0 +1,145 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, pythonRelaxDepsHook
+# propagated build input
+, faiss
+, torch
+, transformers
+, huggingface-hub
+, numpy
+, pyyaml
+, regex
+# optional-dependencies
+, aiohttp
+, fastapi
+, uvicorn
+# , apache-libcloud
+, rich
+, duckdb
+, pillow
+, networkx
+, python-louvain
+, onnx
+, onnxruntime
+, soundfile
+, scipy
+, ttstokenizer
+, beautifulsoup4
+, nltk
+, pandas
+, tika
+, imagehash
+, timm
+, fasttext
+, sentencepiece
+, accelerate
+, onnxmltools
+, annoy
+, hnswlib
+, scikit-learn
+, sentence-transformers
+, croniter
+, openpyxl
+, requests
+, xmltodict
+# native check inputs
+, unittestCheckHook
+}:
+let
+  version = "5.5.1";
+  api = [ aiohttp fastapi uvicorn ];
+  # cloud = [ apache-libcloud ];
+  console = [ rich ];
+  database = [ duckdb pillow ];
+  graph = [ networkx python-louvain ];
+  model = [ onnx onnxruntime ];
+  pipeline-audio = [ onnx onnxruntime soundfile scipy ttstokenizer ];
+  pipeline-data = [ beautifulsoup4 nltk pandas tika ];
+  pipeline-image = [ imagehash pillow timm ];
+  pipeline-text = [ fasttext sentencepiece ];
+  pipeline-train = [ accelerate onnx onnxmltools onnxruntime ];
+  pipeline = pipeline-audio ++ pipeline-data ++ pipeline-image ++ pipeline-text ++ pipeline-train;
+  similarity = [
+    annoy
+    fasttext
+    hnswlib
+    # pymagnitude-lite
+    scikit-learn
+    sentence-transformers
+  ];
+  workflow = [
+    # apache-libcloud
+    croniter
+    openpyxl
+    pandas
+    pillow
+    requests
+    xmltodict
+  ];
+  all = api ++ console ++ database ++ graph ++ model ++ pipeline ++ similarity ++ workflow;
+  optional-dependencies = {
+    inherit api console database graph model pipeline-audio pipeline-image
+      pipeline-text pipeline-train pipeline similarity workflow all;
+  };
+in
+buildPythonPackage {
+  pname = "txtai";
+  inherit version;
+  format = "setuptools";
+
+  disable = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "neuml";
+    repo = "txtai";
+    rev = "v${version}";
+    hash = "sha256-h6TwWzLYfFg5x2QMIstAZ5pkxfHobBU+b4gb0HiayzY=";
+  };
+
+  nativeBuildInputs = [
+    pythonRelaxDepsHook
+  ];
+
+  pythonRemoveDeps = [
+    # We call it faiss, not faiss-cpu.
+    "faiss-cpu"
+  ];
+
+  propagatedBuildInputs = [
+    faiss
+    torch
+    transformers
+    huggingface-hub
+    numpy
+    pyyaml
+    regex
+  ];
+
+  pythonImportsCheck = [ "txtai" ];
+
+  # some tests hang forever
+  doCheck = false;
```

Comment:
You should elaborate why that is.

Reply:
I don't know unfortunately. I wish I had more information, but it just stops and doesn't give a lot of output.
I definitely want to improve this, but I didn't know really what to do.

Reply:
Thanks for the suggestion, it doesn't look like nix liked that

```
> Executing pytestCheckPhase
       > ============================= test session starts =============================
=
       > platform darwin -- Python 3.10.12, pytest-7.2.1, pluggy-1.0.0 -- /nix/store/j7l
vji2dg1bkr7j8das3fak85hbxjzpx-python3-3.10.12/bin/python3.10
       > cachedir: .pytest_cache
       > rootdir: /private/tmp/nix-build-python3.10-txtai-5.5.1.drv-0/source
       > plugins: anyio-3.7.0
       > collected 0 items
       >
       > ============================ no tests ran in 0.04s =============================
       ```

Reply:
The files for the tests have an unusual name, and pytest cannot recognize them.
Looking at the unittest logs, more than half of the tests fail because they try to access huggingface.co.